pennylane/templates/layers/custom_entangler.py

```diff
@@ -212,31 +212,31 @@ def circuit(weights):
 
     # Checks cases where there are coupling parameters
 
-    if type(pattern) == list:
+    if isinstance(pattern, list):
         check_shapes(pattern, [(2,)], msg="Elements of custom 'pattern' must be of shape (2,)") 
-    
+
     if coupling_weights is not None:
 
-       repeat_coupling = check_number_of_layers([coupling_weights])
-       if (repeat_coupling != repeat):
-           raise ValueError("First dimension of `rotation_weights` and `coupling_weights` must be the same")
-
-       if pattern in OPTIONS:
-           expected_shape = (repeat_coupling, n_parameters[pattern])
-           check_shape(
-               coupling_weights, 
-               expected_shape,
-               msg="'coupling_weights' must be of shape {}; got {}" "".format(expected_shape, get_shape(coupling_weights)),
-               )
-
-       elif type(pattern) == list:
-           expected_shape = (repeat_coupling, len(pattern))
-           check_shape(
-               coupling_weights, 
-               expected_shape,
-               msg="'coupling_weights' must be of shape {}; got {}" "".format(expected_shape, get_shape(coupling_weights)),
-               )
-    
+        repeat_coupling = check_number_of_layers([coupling_weights])
+        if repeat_coupling != repeat:
+            raise ValueError("First dimension of `rotation_weights` and `coupling_weights` must be the same")
+
+        if pattern in OPTIONS:
+            expected_shape = (repeat_coupling, n_parameters[pattern])
+            check_shape(
+                coupling_weights,
+                expected_shape,
+                msg="'coupling_weights' must be of shape {}; got {}" "".format(expected_shape, get_shape(coupling_weights)),
+                )
+
+        elif type(pattern) == list:
```

Comment:
I would use ``isinstance``, that is best practice because it also catches types that the object inherits from, if i am not mistaken.